src/pybind/mgr/dashboard/frontend/src/app/shared/pipes/boolean-text.pipe.ts

```diff
@@ -0,0 +1,16 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+import { I18n } from '@ngx-translate/i18n-polyfill';
+
+@Pipe({
+  name: 'booleanText'
+})
+export class BooleanTextPipe implements PipeTransform {
+  constructor(private i18n: I18n) {}
+
+  transform(value: any, truthyText?: string, falsyText?: string): string {
```

Comment:
You can assign the i18n values as parameters if they are not used. I'm also not sure if `value as boolean` has to be used as it not changes the value, `Boolean(value)` would do it.

```diff
diff --git a/src/dmclock b/src/dmclock
index dd48b21100..5c82208802 160000
--- a/src/dmclock
+++ b/src/dmclock
@@ -1 +1 @@
-Subproject commit dd48b2110091e9db2d53a7c8e06d906abb907dbd
+Subproject commit 5c82208802340fe240032f16604b0f9961247849
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/shared/pipes/boolean-text.pipe.ts b/src/pybind/mgr/dashboard/frontend/src/app/shared/pipes/boolean-text.pipe.ts
index a6818fcc75..52b1560a80 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/shared/pipes/boolean-text.pipe.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/shared/pipes/boolean-text.pipe.ts
@@ -8,9 +8,11 @@ import { I18n } from '@ngx-translate/i18n-polyfill';
 export class BooleanTextPipe implements PipeTransform {
   constructor(private i18n: I18n) {}
 
-  transform(value: any, truthyText?: string, falsyText?: string): string {
-    truthyText = truthyText || this.i18n('Yes');
-    falsyText = falsyText || this.i18n('No');
-    return (value as boolean) ? truthyText : falsyText;
+  transform(
+    value: any,
+    truthyText: string = this.i18n('Yes'),
+    falsyText: string = this.i18n('No')
+  ): string {
+    return value ? truthyText : falsyText;
   }
 }
```

Reply:
Done